ENIGMAsystem/SHELL/Platforms/Cocoa/CocoaJoystick.cpp

```diff
@@ -0,0 +1,72 @@
+/** Copyright (C) 2016 Rodrigo Rocha
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include <string>
+using namespace std;
+
+namespace enigma
+{
+  //stub
+}
+
+/* Note: empty stub functions to build Mac executables without errors (and no joystick support, of course) */
+namespace enigma_user
+{
+  bool joystick_load(int id) {
+    return false; //stub
```

Comment:
That would be nice, but unfortunately, that would just warn when this file was compiled (rather than when the method was used). We could do a `show_error` here to let them know that the joystick load is failing because it's unimplemented, but that's not great, either. Don't worry about it. This is a problem with ENIGMA's compiler and the setup of the system, not with this pull request. ☺